### ISSUE TITLE
Training own model makes segmentation worse 

### ISSUE THREAD

**Issue by nordmannth** [MouseLand/cellpose#813]
Hi
Thank you very much for this amazing and very useful tool. 
While training several succesful models in the past, I am currently trying to train a segmentation model for CD45+ immune cells, along with the nuclear staining as an additional channel. Unfortunately, I constantly keep getting 0 masks after training either from pretrained models or even from scratch. Even when taking a single image, annotating the masks and training on this only, I get 0 masks. 

Here the log file, as an example. 

2023-11-04 16:15:17,161 [INFO] training with ['ap1_wA1_t1_m39_c3_z1_l1_o0.tiff']
2023-11-04 16:15:17,161 [INFO] training new model starting from scratch
2023-11-04 16:15:17,161 [INFO] training with chan = 1: red, chan2 = 2: green
2023-11-04 16:15:17,161 [INFO] ** TORCH CUDA version installed and working. **
2023-11-04 16:15:17,161 [INFO] >>>> using GPU
GUI_INFO: name of new model: newscratch_red_green
2023-11-04 16:15:18,140 [INFO] computing flows for labels
100%|████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.60it/s]
2023-11-04 16:15:19,219 [INFO] >>>> median diameter set to = 30
2023-11-04 16:15:19,219 [INFO] >>>> mean of training label mask diameters (saved to model) 46.598
2023-11-04 16:15:19,219 [INFO] >>>> training network with 2 channel input <<<<
2023-11-04 16:15:19,219 [INFO] >>>> LR: 0.00100, batch_size: 8, weight_decay: 0.00010
2023-11-04 16:15:19,219 [INFO] >>>> ntrain = 1
2023-11-04 16:15:19,219 [INFO] >>>> nimg_per_epoch = 8
2023-11-04 16:15:20,937 [INFO] Epoch 0, Time  1.7s, Loss 0.6223, LR 0.0000
2023-11-04 16:15:22,106 [INFO] saving network parameters to E:/ThNo/test\models/newscratch_red_green
2023-11-04 16:15:27,505 [INFO] Epoch 5, Time  8.3s, Loss 0.6258, LR 0.0006
2023-11-04 16:15:34,261 [INFO] Epoch 10, Time 15.0s, Loss 0.6523, LR 0.0010
2023-11-04 16:15:47,531 [INFO] Epoch 20, Time 28.3s, Loss 0.5860, LR 0.0010
2023-11-04 16:16:00,523 [INFO] Epoch 30, Time 41.3s, Loss 0.5362, LR 0.0010
2023-11-04 16:16:13,787 [INFO] Epoch 40, Time 54.6s, Loss 0.5625, LR 0.0010
2023-11-04 16:16:25,781 [INFO] saving network parameters to E:/ThNo/test\models/newscratch_red_green
E:/ThNo/test\models/newscratch_red_green copied to models folder C:\Users\bias\.cellpose\models
newscratch_red_green removed from custom model list
(3008, 4096, 4)
GUI_INFO: auto-adjust enabled, computing saturation levels
GUI_INFO: loaded in previous changes
GUI_INFO: 154 masks found in E:/ThNo/test\ap1_wA1_t1_m39_c3_z1_l1_o0_seg.npy
2023-11-04 16:16:27,632 [INFO] >>>> diameter set to diam_labels ( =  46.598 )
2023-11-04 16:16:28,250 [INFO] >>>> loading model C:\Users\bias\.cellpose\models\newscratch_red_green
2023-11-04 16:16:28,250 [INFO] ** TORCH CUDA version installed and working. **
2023-11-04 16:16:28,250 [INFO] >>>> using GPU
2023-11-04 16:16:28,412 [INFO] >>>> model diam_mean =  30.000 (ROIs rescaled to this size during training)
2023-11-04 16:16:28,412 [INFO] >>>> model diam_labels =  46.598 (mean diameter of training ROIs)
2023-11-04 16:16:38,591 [INFO] 0 cells found with model in 10.394 sec
GUI_INFO: 0 masks found
GUI_INFO: creating cellcolors and drawing masks
2023-11-04 16:16:38,927 [INFO] !!! computed masks for ap1_wA1_t1_m39_c3_z1_l1_o0.tiff from new model !!!
2023-11-04 16:22:17,750 [INFO] training with ['ap1_wA1_t1_m39_c3_z1_l1_o0.tiff']
2023-11-04 16:22:17,750 [INFO] training new model starting at model TN2
2023-11-04 16:22:17,750 [INFO] training with chan = 1: red, chan2 = 2: green
2023-11-04 16:22:17,762 [INFO] >> TN2 << model set to be used
2023-11-04 16:22:17,762 [INFO] ** TORCH CUDA version installed and working. **
2023-11-04 16:22:17,762 [INFO] >>>> using GPU
2023-11-04 16:22:17,924 [INFO] >>>> model diam_mean =  30.000 (ROIs rescaled to this size during training)
GUI_INFO: name of new model: newTN2_red_green
2023-11-04 16:22:18,875 [INFO] computing flows for labels
100%|████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.39it/s]
2023-11-04 16:22:20,108 [INFO] >>>> median diameter set to = 30
2023-11-04 16:22:20,109 [INFO] >>>> mean of training label mask diameters (saved to model) 46.598
2023-11-04 16:22:20,109 [INFO] >>>> training network with 2 channel input <<<<
2023-11-04 16:22:20,109 [INFO] >>>> LR: 0.10000, batch_size: 8, weight_decay: 0.00010
2023-11-04 16:22:20,109 [INFO] >>>> ntrain = 1
2023-11-04 16:22:20,109 [INFO] >>>> nimg_per_epoch = 8
2023-11-04 16:22:21,827 [INFO] Epoch 0, Time  1.7s, Loss 3.8237, LR 0.0000
2023-11-04 16:22:23,024 [INFO] saving network parameters to E:/ThNo/test\models/newTN2_red_green
2023-11-04 16:22:28,608 [INFO] Epoch 5, Time  8.5s, Loss 0.6766, LR 0.0556
2023-11-04 16:22:35,640 [INFO] Epoch 10, Time 15.5s, Loss 0.1530, LR 0.1000
2023-11-04 16:22:49,218 [INFO] Epoch 20, Time 29.1s, Loss 0.0875, LR 0.1000
2023-11-04 16:23:02,508 [INFO] Epoch 30, Time 42.4s, Loss 0.1008, LR 0.1000
2023-11-04 16:23:16,055 [INFO] Epoch 40, Time 55.9s, Loss 0.2135, LR 0.1000
2023-11-04 16:23:28,280 [INFO] saving network parameters to E:/ThNo/test\models/newTN2_red_green
E:/ThNo/test\models/newTN2_red_green copied to models folder C:\Users\bias\.cellpose\models
(3008, 4096, 4)
GUI_INFO: auto-adjust enabled, computing saturation levels
GUI_INFO: loaded in previous changes
GUI_INFO: 154 masks found in E:/ThNo/test\ap1_wA1_t1_m39_c3_z1_l1_o0_seg.npy
2023-11-04 16:23:30,204 [INFO] >>>> diameter set to diam_labels ( =  46.598 )
2023-11-04 16:23:30,830 [INFO] >>>> loading model C:\Users\bias\.cellpose\models\newTN2_red_green
2023-11-04 16:23:30,830 [INFO] ** TORCH CUDA version installed and working. **
2023-11-04 16:23:30,830 [INFO] >>>> using GPU
2023-11-04 16:23:30,976 [INFO] >>>> model diam_mean =  30.000 (ROIs rescaled to this size during training)
2023-11-04 16:23:30,976 [INFO] >>>> model diam_labels =  46.598 (mean diameter of training ROIs)
2023-11-04 16:23:39,289 [INFO] 0 cells found with model in 8.523 sec
GUI_INFO: 0 masks found
GUI_INFO: creating cellcolors and drawing masks
2023-11-04 16:23:39,642 [INFO] !!! computed masks for ap1_wA1_t1_m39_c3_z1_l1_o0.tiff from new model !!!

Here is the same image, and the .npy file. R = membrane marker CD45 ; G = nuclear channel; B = Cytokeratin+ cells (keratinocytes; irrelevant for this segmentation)

https://fastupload.io/ROXgbEEl6OsCK90/file

 I normally use more than 50 epoch, but as the result has been the same I lowered this for debugging. Any suggestions as to why this is occuring? Any help would be extremely appreciated. 

Thanks alot!
Best, 
Thierry

**Comment by carsen-stringer** [comment]
closing due to inactivity
